process/smartContract/process.go

```diff
@@ -1315,12 +1334,55 @@ func createBaseSCR(
 	return result
 }
 
+func addVMOutputResultsToSCR(vmOutput *vmcommon.VMOutput, result *smartContractResult.SmartContractResult) {
+	result.CallType = vmcommon.AsynchronousCallBack
+	result.GasLimit = vmOutput.GasRemaining
+	result.Data = []byte("@" + core.ConvertToEvenHex(int(vmOutput.ReturnCode)))
+	addReturnDataToSCR(vmOutput, result)
+}
+
+func createAsyncCallBackSCRFromVMOutput(
+	vmOutput *vmcommon.VMOutput,
+	tx data.TransactionHandler,
+	txHash []byte,
+) *smartContractResult.SmartContractResult {
+	scr := &smartContractResult.SmartContractResult{
+		Value:          big.NewInt(0),
+		RcvAddr:        tx.GetSndAddr(),
```

Comment:
The scr receiver is actually the original sender right?

Reply:
the sender of the transaction. as this was an asyncCall - the sender and receiver switch places.